assembly/Int/CompactInt.ts

```diff
@@ -0,0 +1,58 @@
+import { Bytes } from "../utils/Bytes";
+import { Codec } from "../interfaces/Codec";
+import { BIT_LENGTH } from "../utils/Bytes";
+import { BytesBuffer } from "../utils/BytesBuffer";
+
+/** Representation for a Int8 value in the system. */
+export class CompactInt implements Codec {
+
+    public readonly value: i64;
+    protected bitLength: i32 = BIT_LENGTH.INT_8;
+
+    constructor (value: i64) {
+        this.value = value;
+    }
+
+    /**
+    * @description  Encodes the value as u8[] as per the SCALE codec specification
+    */
+    public toU8a (): u8[] {
+        const bytesBuffer = new BytesBuffer();
+        bytesBuffer.encodeLength(this.value);
+
+        return bytesBuffer.bytes;
+    }
+
+    /**
+    * @description Returns the string representation of the value
+    */
+    toString (): string {
+        return this.value.toString();
+    }
+
+    /**
+     * @description The length of Int when the value is encoded
+     */
+    public encodedLength (): i32 {
+        return this.bitLength;
+    }
+
+    /**
+     * @description CompactInt could be 1, 2, 4, 8 bytes. Set bytes dynamically
+     */
+    public setEncodedLength (encodedLength: i32): void {
```

Comment:
I think that we should not set the length from the outside. The idea behind the `encodedLength` is that we instantiate a native object and we can compute what the potential `encodedLength` would be if we execute `toU8a`.
That being said, I think that once we create new instance of the `CompactInt`, in its constructor we must figure out what the encoded length will be by:
```JavaScript
if (i < 1 << 6) -> 1
if (i < i << 14) -> 2
if (i < i << 30) -> 4
otherwise -> 8
```